### PR TITLE
Introduce Show Logs Command

### PR DESCRIPTION
**Pull request by tchaton**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning-hpo/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?

Fixes # (issue).

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: lightning_hpo/controllers/sweep.py [hunk 1]
```diff
@@ -156,6 +156,22 @@ def stop_experiment(self, config: StopExperimentConfig):
                     return f"The current experiment `{experiment.name}` has been stopped."
         return f"The current experiment `{config.name}` doesn't exist."
 
+    def show_logs(self) -> List[ShowLogsConfig]:
+        data = []
+        for sweep in self.db.select_all():
+            if sweep.sweep_id in self.r:
+                works = self.r[sweep.sweep_id].works()
+                if sweep.algorithm:
+                    data.append(ShowLogsConfig(name=sweep.sweep_id, components=[w.name for w in works]))
```
Comment (lantiga) [inline lightning_hpo/controllers/sweep.py:165]: Wouldn't this get tensorboard as well?

File: lightning_hpo/commands/logs/show.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            "--name", required=True, type=str, help="The name of the experiment or Sweep to show logs from."
+        )
+        parser.add_argument("-f", "--follow", action="store_true", default=False)
+        hparams = parser.parse_args()
+
+        logs_config: List[ShowLogsConfig] = self.invoke_handler()
+
+        components = []
+        sweeps = [c["name"] for c in logs_config if len(c["components"]) > 1]
+        experiments = [c["name"] for c in logs_config if len(c["components"]) == 1]
```
Comment (lantiga) [inline lightning_hpo/commands/logs/show.py:32]: I'm not sure I understand this, isn't it getting names twice from the same collection?

**Review by lantiga — CHANGES_REQUESTED** [review]

File: lightning_hpo/commands/logs/show.py (hunk 2 above)
Comment (tchaton) [reply lightning_hpo/commands/logs/show.py:32]: This won't happen. You can enter either a sweep name or experiment name. So, there won't be collision.

File: lightning_hpo/controllers/sweep.py (hunk 1 above)
Comment (tchaton) [reply lightning_hpo/controllers/sweep.py:165]: No, I didn't add support for tensorboard as this wasn't within provided API. Right now, supporting only sweeps and experiments.

File: lightning_hpo/commands/logs/show.py (hunk 2 above)
Comment (tchaton) [reply lightning_hpo/commands/logs/show.py:32]: Unless you want to enable users to pass a collection of names. In this case, we simply add list(set(components))

**Review by ethanwharris — APPROVED** [review]

File: lightning_hpo/commands/logs/show.py (hunk 2 above)
Comment (lantiga) [reply lightning_hpo/commands/logs/show.py:32]: That's right, but then the message below 
```
f"The provided name {hparams.name} wasn't found."
f" Here are the sweeps {sweeps} and experiments {experiments}."
```
needs to be fixed.

File: lightning_hpo/controllers/sweep.py (hunk 1 above)
Comment (lantiga) [reply lightning_hpo/controllers/sweep.py:165]: That's totally fine, I definitely wasn't looking for tensorboard logs :-) 
My question was: when querying the works, wouldn't tensorboard come up when using the experiment name? (I can't recall the naming of tensorboard works)

Comment (tchaton) [reply lightning_hpo/controllers/sweep.py:165]: No, it isn't within another machine.

File: lightning_hpo/commands/logs/show.py (hunk 2 above)
Comment (tchaton) [reply lightning_hpo/commands/logs/show.py:32]: It should be all good. We are filtering experiments from sweeps, so the error message should be good.

**Review by lantiga — APPROVED** [review]

**Pull request merged**